crates/solver/src/solver/single_order_solver.rs

```diff
@@ -202,6 +202,115 @@ impl SingleOrderSolver {
         }
     }
 
+    async fn solve_instrumented(
+        &self,
+        Auction {
+            orders,
+            balances,
+            external_prices,
+            gas_price,
+            deadline,
+            ..
+        }: Auction,
+    ) -> Result<Vec<Settlement>> {
+        let orders = super::balance_and_convert_orders(
+            self.ethflow_contract,
+            &self.order_converter,
+            balances,
+            orders,
+        );
+        let mut orders =
+            get_prioritized_orders(&orders, &external_prices, &self.order_prioritization_config);
+        tracing::trace!(?orders, "prioritized orders");
+
+        let mut settlements = Vec::new();
+        let (tx, mut rx) = mpsc::unbounded_channel::<SingleOrderSettlement>();
+
+        let solve = async {
+            while let Some(order) = orders.pop_front() {
+                let span = tracing::info_span!("solve", id =? order.id);
```

Comment:
Since we have a parent `solve` span, I would propose this span to be called:
```suggestion
                let span = tracing::info_span!("order", id =? order.id);
```